### DISCUSSION TITLE
Empty supp_dict (image_classification.py)

### DISCUSSION THREAD

**Discussion by andreamigliorati** [yoshitomo-matsubara/sc2-benchmark#144] [Q&A]
Hi, I want to perform image classification and train my own models, I added the train keys to the config file to do that. However I have a problem with the `train_one_epoch` method. When computing the loss (`loss = training_box(sample_batch, targets, supp_dict)`) I get an error which appears to be caused by an empty `supp_dict`.

```
File "/media/RAID/migliorati/coding_machines/sc2-benchmark-main/script/task/image_classification.py", line 209, in main
    train(teacher_model, student_model, dataset_dict, ckpt_file_path, device, device_ids, distributed, config, args)
  File "/media/RAID/migliorati/coding_machines/sc2-benchmark-main/script/task/image_classification.py", line 160, in train
    train_one_epoch(training_box, aux_module, bottleneck_updated, device, epoch, log_freq)
  File "/media/RAID/migliorati/coding_machines/sc2-benchmark-main/script/task/image_classification.py", line 73, in train_one_epoch
    loss = training_box(sample_batch, targets, supp_dict)
```
Before being called in `loss = training_box(sample_batch, targets, supp_dict)`, `supp_dict` is created here:

```
    for sample_batch, targets, supp_dict in \
            metric_logger.log_every(training_box.train_data_loader, log_freq, header):
        if isinstance(sample_batch, torch.Tensor):
            sample_batch = sample_batch.to(device)

        if isinstance(targets, torch.Tensor):
            targets = targets.to(device)
```
which in my case gives an empty `supp_dict`, and therefore a missing key (`KeyError: 'backbone.bottleneck_layer'`). I am launching my training with `bash script/neural_input_compression/ilsvrc2012-image_classification.sh factorized_prior-resnet50 8`, and as mentioned I created a train key in the config file so I am not sure what I'm doing wrong. Thanks in advance!

**Comment by yoshitomo-matsubara** [comment]
Hi @andreamigliorati 

Can you share your yaml file here?

**Reply by yoshitomo-matsubara** [reply]
Please use "Reply" to organize discussions.

Empty `supp_dict` doesn't matter in this case. You're trying to train **Factorized Prior model + ResNet-50** using a configuration to train **a modified ResNet-50 model with a non-entropic bottleneck with a pretrained ResNet-50 as a teacher for GHND + BQ (bottleneck quantization)**.

I recommend that you decide 1) how you want to train 2) what model. Since you'll also want to know about designs of the models, references in the SC2 benchmark paper should also help you. If you just want to train some random models, I suggest starting with config files under `supervised_compression/`

**Comment by andreamigliorati** [comment]
Yes of course, here it is. Thanks for the help, again apologies in advance if I'm asking something trivial but I just can't seem to make it work for now.
[factorized_prior-resnet50.zip](https://github.com/user-attachments/files/16196289/factorized_prior-resnet50.zip)
